### PR TITLE
Register new snapshots

### PR DESCRIPTION
**Pull request by alexcrichton**
Lots of cruft to remove!

### CODE REVIEW

**Comment by alexcrichton** [comment]
(or anyone else)

**Comment by arielb1** [comment]
This version has a performance regression DO NOT SNAPSHOT UNTIL IT IS FIXED!

File: src/librustc_driver/lib.rs [hunk 1]
```diff
@@ -105,22 +101,8 @@ pub mod target_features;
...
-    use std::raw::TraitObject;
-    unsafe {
-        let TraitObject { vtable, .. } =
-            mem::transmute::<&for<'s> CrateStore<'s>, TraitObject>(&*a);
-        mem::transmute(TraitObject {
-            data: mem::transmute(a),
-            vtable: vtable
-        })
-    }
+pub fn cstore_to_cratestore(a: Rc<CStore>) -> Rc<for<'s> CrateStore<'s>> {
```
Comment (arielb1) [inline src/librustc_driver/lib.rs]: just kill the function. it has no reason for existing.

**Comment by alexcrichton** [comment]
@arielb1 oh hm, is there an issue for that? Depending on how much the snapshot is desired it may be worth making a snapshot before the regression (or just going ahead if the regression isn't that bad anyway)

**Comment by dotdash** [comment]
@alexcrichton I guess this might be about #30242 which apparently introduced a perf regression. There's no issue for that yet AFAICT.

File: src/liballoc/rc.rs [hunk 2]
```diff
@@ -197,7 +195,6 @@ impl<T: ?Sized> !marker::Send for Rc<T> {}
 impl<T: ?Sized> !marker::Sync for Rc<T> {}
 
 // remove cfg after new snapshot
```
Comment (gereeter) [inline src/liballoc/rc.rs]: Comment should be removed?

File: src/liballoc/rc.rs [hunk 3]
```diff
@@ -724,7 +721,6 @@ impl<T: ?Sized> !marker::Send for Weak<T> {}
 impl<T: ?Sized> !marker::Sync for Weak<T> {}
 
 // remove cfg after new snapshot
```
Comment (gereeter) [inline src/liballoc/rc.rs]: Comment should be removed?

File: src/liballoc/rc.rs (hunk 2 above)
Comment (alexcrichton) [inline src/liballoc/rc.rs]: Oops, it indeed can

**Comment by pnkfelix** [comment]
I think that #30368 is @arielb1 's PR to address the performance regression.

**Comment by alexcrichton** [comment]
Rebased and updated with new snapshots that include #30368

**Comment by nikomatsakis** [comment]
@bors r+ p=1

Travis has been kind of flaky lately, eh? Is there a reason for that?

**Comment by alexcrichton** [comment]
@bors: r=nikomatsakis 34a75ea7abc6472498e91f7e7f2d7484429b7aa9

Actually travis seems to be pretty reliable for me at least as that's a test I needed to fix!

**Pull request merged**